lib/streamlit/MediaFileManager.py

```diff
@@ -41,48 +48,62 @@ def _get_file_id(data, mimetype=None):
     filehash.update(bytes(mimetype.encode("utf-8")))
     return filehash.hexdigest()
 
-    #return hashlib.sha224(bytes(mimetype.encode("utf-8")) + data).hexdigest()
-
 
 class MediaFile(object):
     """Abstraction for audiovisual/image file objects."""
 
-    def __init__(self, file_id=None, content=None, mimetype=None):
+    def __init__(self, file_id=None, content=None, mimetype=None, refcount=1):
         self.file_id = file_id
         self.content = content
         self.mimetype = mimetype
+        self.refcount = refcount
 
     @property
     def url(self):
         return "{}/{}.{}".format(
             STATIC_MEDIA_ENDPOINT, self.file_id, self.mimetype.split("/")[1]
         )
 
+    @property
+    def id(self):
+        return self.file_id
+
 
 class MediaFileManager(object):
     """In-memory file manager for MediaFile objects."""
 
     def __init__(self):
-        self._files = {}
-
-    def clear(self):
-        """Deletes all files from the file manager. """
-        self._files.clear()
+        self._files = collections.defaultdict()
```

Comment:
This `defaultdict` doesn't have any default, so you might as well use a plain old dict here instead: `{}`